src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs

```diff
@@ -112,84 +118,103 @@ private void OnDestroy()
 #endif
             if (_isRecording)
             {
-                HandleRecordingSaving();
+                KeyboardInputActionObserver.GetInstance()?.StopRecording();
+                _mouseObserver.ClearBuffer();
+                _isRecording = false;
+                _ = HandleEndRecording(_tickNumber, _startTime, DateTime.Now, _currentGameplaySessionDataDirectoryPrefix, _currentGameplaySessionScreenshotsDirectoryPrefix, _currentGameplaySessionThumbnailPath);
```

Comment:
Nit, non-blocking: we are passing along a lot of class variables that can just be referenced in the helper function itself.

Reply:
no they can't... the very reason I changed this is that we are calling async functions and were relying on class level variables which very well may change while that async is running

Reply:
🤦 my sincerest apologies, of course... this train ride debacle this morning has me more scrambled than I like